### PR TITLE
added condition for debian 9 default ipv4  interfaces

### LINKED ISSUE #177: INTERFACES default option deprecated on Debian 9
## Affected Puppet, Ruby, OS and module versions/distributions

- Puppet: PE 2017.3
- Ruby: PE ruby
- Distribution: Debian 9 (stretch)
- Module version: 3.2.0

## How to reproduce (e.g Puppet code you use)
Configure IPv4 range to be served from a Debian 9 system.

## What are you seeing
/etc/default/isc-dhcp-server contains an INTERFACES variable. Latest versions appear to have split that up between INTERFACESv4 and INTERFACESv6 so one may server each on differing interfaces as required.

## What behaviour did you expect instead
Expected running servive, but systemd sees it as a failure.

## Output log
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:20 desktop systemd[1]: Stopping LSB: DHCP server...
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:20 desktop isc-dhcp-server[16583]: Stopping ISC DHCPv4 server: dhcpd.
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:20 desktop systemd[1]: Stopped LSB: DHCP server.
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:20 desktop systemd[1]: Starting LSB: DHCP server...
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:20 desktop isc-dhcp-server[16596]: DHCPv4 interfaces are no longer set by the INTERFACES variable in
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:20 desktop isc-dhcp-server[16596]: /etc/default/isc-dhcp-server.  Please use INTERFACESv4 instead.
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:20 desktop isc-dhcp-server[16596]: Migrating automatically for now, but this will go away in the futu
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:20 desktop isc-dhcp-server[16596]: Launching IPv4 server only.
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:20 desktop dhcpd[16607]: Internet Systems Consortium DHCP Server 4.3.5
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:20 desktop dhcpd[16607]: Copyright 2004-2016 Internet Systems Consortium.
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:20 desktop dhcpd[16607]: All rights reserved.
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:20 desktop dhcpd[16607]: For info, please visit https://www.isc.org/software/dhcp/
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:20 desktop dhcpd[16607]: Wrote 0 deleted host decls to leases file.
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:20 desktop dhcpd[16607]: Wrote 0 new dynamic host decls to leases file.
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:20 desktop dhcpd[16607]: Wrote 29 leases to leases file.
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:22 desktop isc-dhcp-server[16596]: Starting ISC DHCPv4 server: dhcpdcheck syslog for diagnostics. ...
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:22 desktop isc-dhcp-server[16596]:  failed!
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:22 desktop systemd[1]: isc-dhcp-server.service: Control process exited, code=exited status=1
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:22 desktop systemd[1]: Failed to start LSB: DHCP server.
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:22 desktop systemd[1]: isc-dhcp-server.service: Unit entered failed state.
Mar 11 20:46:22 desktop puppet-agent[16440]: (/Stage[main]/Dhcp/Service[isc-dhcp-server]) Mar 11 20:46:22 desktop systemd[1]: isc-dhcp-server.service: Failed with result 'exit-code'.

### PR DESCRIPTION
**Pull request by tiernap**
#### Pull Request (PR) description

INTERFACES default option deprecated on Debian 9. replaced with basic condition (INTERFACESv4 only)

#### This Pull Request (PR) fixes the following issues

Fixes #177

### CODE REVIEW

File: templates/debian/default_isc-dhcp-server [hunk 1]
```diff
@@ -8,4 +8,8 @@
 
 # On what interfaces should the DHCP server (dhcpd) serve DHCP requests?
 #       Separate multiple interfaces with spaces, e.g. "eth0 eth1".
+<% if @lsbmajdistrelease >= '9' -%>
```
Comment (ekohl) [inline templates/debian/default_isc-dhcp-server]: This is also used for Ubuntu so I think this needs a little bit more attention.

File: templates/debian/default_isc-dhcp-server [hunk 2]
```diff
@@ -8,4 +8,8 @@
 
 # On what interfaces should the DHCP server (dhcpd) serve DHCP requests?
 #       Separate multiple interfaces with spaces, e.g. "eth0 eth1".
+<% if @lsbmajdistrelease >= '9' && @lsbdistid == 'Debian' -%>
```
Comment (ekohl) [inline templates/debian/default_isc-dhcp-server]: Personally I prefer the facts `operatingsystem` and `operatingsystemmajrelease`. It looks like facterdb doesn't know the lsb versions either which causes the tests to fail.

Comment (tiernap) [reply templates/debian/default_isc-dhcp-server]: thank you . this is changed

**Comment by tiernap** [comment]
is there something to modify to pass the tests?

thank you

**Comment by ekohl** [comment]
It looks like the tests are still old style without facter so you need to provide the facts.

**Comment by tiernap** [comment]
i have added default facts but it is still failing. do you have any suggestion? 

thank you for your help

**Pull request closed without merging**